### PR TITLE
feat: store message history

### PR DESCRIPTION
**Pull request by jbrunton**

### CODE REVIEW

File: server/src/app/message.ts [hunk 1]
```diff
@@ -37,13 +38,15 @@ export const handleMessage = async (
   dispatcher: Dispatcher
 ) => {
   if (isCommand(message)) {
-    const response = await processCommand(message, dispatcher);
```
Comment (jbrunton) [inline server/src/app/message.ts:40]: Extract this code to a multiplexer use case.

File: server/src/app/rooms/fs-room-repository.ts [hunk 2]
```diff
@@ -69,4 +77,25 @@ export class FsRoomRepository implements RoomRepository {
   async renameRoom(params: RenameRoomParams): Promise<Room> {
     return this.db.updateRoom(params);
   }
+
+  async saveMessage(message: Message): Promise<Message> {
```
Comment (jbrunton) [inline server/src/app/rooms/fs-room-repository.ts]: Give this a "DraftMessage" (or IncomingMessage) type which doesn't include id field.

File: server/src/app/rooms/fs-room-repository.ts [hunk 3]
```diff
@@ -69,4 +77,25 @@ export class FsRoomRepository implements RoomRepository {
...
+    const newMessage = {
+      ...message,
+      id,
+    };
+    this.db.createMessage(newMessage);
+    debug.room("saved message: %O", newMessage);
+    return newMessage;
+  }
+
+  async getMessageHistory(roomId: string): Promise<Message[]> {
```
Comment (jbrunton) [inline server/src/app/rooms/fs-room-repository.ts:93]: Add tests for changes to this repository

**Comment by jbrunton-ci-minion** [comment]
#### :tropical_drink: `up` on message-history

   <details>
   <summary>Click to expand Pulumi report</summary>

   ```
   Updating (message-history)

View Live: https://app.pulumi.com/jbrunton/chat-demo/message-history/updates/5


   pulumi:pulumi:Stack chat-demo-message-history running 
   pulumi:pulumi:Stack chat-demo-message-history running stack config: {
~  digitalocean:index:App chat-demo-message-history updating [diff: ~spec]
@ Updating......
~  digitalocean:index:App chat-demo-message-history updated [diff: ~spec]
   pulumi:pulumi:Stack chat-demo-message-history  1 message

Diagnostics:
 pulumi:pulumi:Stack (chat-demo-message-history):
   stack config: {
    "stackName": "message-history",
    "appName": "chat-demo-message-history",
    "environment": "development",
    "tag": "HEAD-2ed010b",
    "protect": false,
    "domain": "message-history.chat-demo.dev.jbrunton-do.com",
    "publicUrl": "https://message-history.chat-demo.dev.jbrunton-do.com",
    "rootDomain": "jbrunton-do.com",
    "specId": "chat-demo-message-history/HEAD-2ed010b/c3f3c59b"
   }

Outputs:
   publicUrl: "https://message-history.chat-demo.dev.jbrunton-do.com"

Resources:
   ~ 1 updated
   1 unchanged

Duration: 1m18s


   ```
   
   </details>

**Pull request merged**